resolver/addressable_resolver.go

```diff
@@ -63,50 +63,31 @@ func NewURIResolver(ctx context.Context, callback func(types.NamespacedName)) *U
 }
 
 // URIFromDestination resolves a Destination into a URI string.
-func (r *URIResolver) URIFromDestination(dest apisv1alpha1.Destination, parent interface{}) (string, error) {
-	var deprecatedObjectReference *corev1.ObjectReference
-	if dest.DeprecatedAPIVersion == "" && dest.DeprecatedKind == "" && dest.DeprecatedName == "" && dest.DeprecatedNamespace == "" {
-		deprecatedObjectReference = nil
-	} else {
-		deprecatedObjectReference = &corev1.ObjectReference{
-			Kind:       dest.DeprecatedKind,
-			APIVersion: dest.DeprecatedAPIVersion,
-			Name:       dest.DeprecatedName,
-			Namespace:  dest.DeprecatedNamespace,
-		}
-	}
-	if dest.Ref != nil && deprecatedObjectReference != nil {
-		return "", fmt.Errorf("ref and [apiVersion, kind, name] can't be both present")
-	}
-	var ref *corev1.ObjectReference
-	if dest.Ref != nil {
-		ref = dest.Ref
-	} else {
-		ref = deprecatedObjectReference
+func (r *URIResolver) URIFromDestination(ctx context.Context, namespace string, dest apisv1alpha1.Destination, parent interface{}) (string, error) {
+	err := dest.Validate(ctx)
+	if err != nil {
+		return "", errors.New(err.Error())
 	}
+	ref := dest.GetRef()
 	if ref != nil {
+		// Do not modify original copy w/o namespace.
+		ref = ref.DeepCopy()
```

Comment:
I don't think this is needed, dest is not a pointer, so it is local to this function.